clients/rust/tests/sync_sol_stake.rs

```diff
@@ -0,0 +1,637 @@
+#![cfg(feature = "test-sbf")]
+
+mod setup;
+
+use paladin_stake_program_client::{
+    accounts::{Config, SolStakerStake, ValidatorStake},
+    errors::PaladinStakeProgramError,
+    instructions::SyncSolStakeBuilder,
+};
+use setup::{
+    config::ConfigManager,
+    new_program_test, setup,
+    sol_staker_stake::SolStakerStakeManager,
+    stake::{create_stake_account, deactivate_stake_account},
+    validator_stake::ValidatorStakeManager,
+    vote::add_vote_account,
+};
+use solana_program_test::tokio;
+use solana_sdk::{
+    account::{Account, AccountSharedData},
+    instruction::InstructionError,
+    pubkey::Pubkey,
+    signature::{Keypair, Signer},
+    stake::state::{Authorized, Lockup},
+    transaction::Transaction,
+};
+
+#[tokio::test]
+async fn sync_sol_stake_when_deactivating() {
+    let mut context = setup().await;
+
+    // Given a config, validator stake and sol staker stake accounts with 5 SOL staked.
+
+    let config_manager = ConfigManager::new(&mut context).await;
+    let validator_stake_manager =
+        ValidatorStakeManager::new(&mut context, &config_manager.config).await;
+    let sol_staker_staker_manager = SolStakerStakeManager::new(
+        &mut context,
+        &config_manager.config,
+        &validator_stake_manager.stake,
+        &validator_stake_manager.vote,
+        5_000_000_000, // 5 SOL staked
+    )
+    .await;
+
+    // And the SOL staker stake and validator stake accounts are correctly synced.
+
+    let account = get_account!(context, sol_staker_staker_manager.stake);
+    let stake_account = SolStakerStake::from_bytes(account.data.as_ref()).unwrap();
+
+    assert_eq!(stake_account.lamports_amount, 5_000_000_000);
+
+    let account = get_account!(context, validator_stake_manager.stake);
+    let validator_stake_account = ValidatorStake::from_bytes(account.data.as_ref()).unwrap();
+    assert_eq!(
+        validator_stake_account.total_staked_lamports_amount,
+        5_000_000_000
+    );
+
+    // And we deactivate the stake.
+
+    deactivate_stake_account(
+        &mut context,
+        &stake_account.sol_stake,
+        &sol_staker_staker_manager.authority,
+    )
+    .await;
```

Comment:
This test hasn't warped to stake activation time, so the deactivation is instantaneous, and it didn't catch the `deactivating` issue brought up in the processor. Can you add a warp to make sure that the stake is active before deactivating it?